### PR TITLE
Feature/add typeid

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
#106 

**This PR will result in the following new package version:**

- v0.10.0 - Adding columns to some tables that affect transform models.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
- Because of the [May 2023 connector update](https://fivetran.com/docs/applications/hubspot/changelog#may2023) `type_id` was added to `DEAL_COMPANY` and `DEAL_CONTACT`. These tables are not pulled into the transform package.
  - I also added column-combo tests for these tables since the update states these are meant to form a composite primary key.

- Also resulting from the connector update, columns `updated_at` and `created_at` have been added to the following sources and their corresponding staging models: 
  - `DEAL_PIPELINE`
  - `DEAL_PIPELINE_STAGE`
  - `TICKET_PIPELINE`
  - `TICKET_PIPELINE_STAGE`
- These columns do get pulled into the transform package, for which I opened a PR on the transform.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (see ticket for vars used)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages. 
  - They will affect transform, which I have opened a PR for in the transform.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

- Validated using data provided in ticket and added tests mentioned above. This was a straight add of columns added by the connector. Ran the published package on our dataset, ran the updated package, and confirm the columns were added with no change in row count/grain.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [ ] ~DECISIONLOG updates have been updated (if applicable)~
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🐈

### CODE REVIEW

File: models/stg_hubspot__ticket.yml [hunk 1]
```diff
@@ -158,3 +136,7 @@ models:
         description: Reference to the object type.
       - name: ticket_pipeline_id
         description: Reference to the pipeline.
+      - name: ticket_pipeline_stage_created_at
+        description: A timestamp representing when the record was created.
+      - name: ticket_pipeline_stage_updated_at
+        description: A timestamp representing when the record was updated.
```
Comment (fivetran-jamie) [inline models/stg_hubspot__ticket.yml:142]: looks like the ticket_pipeline_stage and ticket_pipeline fields got mixed up in here

File: models/stg_hubspot__ticket.yml [hunk 2]
```diff
@@ -140,6 +114,10 @@ models:
         description: The ID of the pipeline stage.
       - name: ticket_state
         description: State of the ticket.
+      - name: ticket_pipeline_created_at
+        description: A timestamp representing when the record was created.
+      - name: ticket_pipeline_updated_at
+        description: A timestamp representing when the record was updated.
```
Comment (fivetran-jamie) [inline models/stg_hubspot__ticket.yml:120]: looks like the ticket_pipeline_stage and ticket_pipeline fields got mixed up in here

**Review by fivetran-jamie — COMMENTED** [review]
looks great! just a couple of documentation comments and random questions

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,20 @@
+# dbt_hubspot_source v0.10.0
+## 🚨 Breaking Changes 🚨
+- In the [May 2023 connector update](https://fivetran.com/docs/applications/hubspot/changelog#may2023) `type_id` was added to sources `DEAL_COMPANY` and `DEAL_CONTACT` as a part of the composite primary key for these tables. This column has been adding to the corresponding staging models. ([PR #100](https://github.com/fivetran/dbt_hubspot_source/pull/100))
+- Updated tests for these tables with `type_id` as part of the primary key. ([PR #100](https://github.com/fivetran/dbt_hubspot_source/pull/100))
+- Also resulting from the connector update, columns `updated_at` and `created_at` have been added to the following sources and their corresponding staging models: ([PR #100](https://github.com/fivetran/dbt_hubspot_source/pull/100))
+  - `DEAL_PIPELINE`
+  - `DEAL_PIPELINE_STAGE`
+  - `TICKET_PIPELINE`
+  - `TICKET_PIPELINE_STAGE`
+- Updated docs with these changes and a little housekeeping. ([PR #100](https://github.com/fivetran/dbt_hubspot_source/pull/100))
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Yes, thank you. Updated!

File: integration_tests/seeds/deal_pipeline_data.csv [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
-mp_pipeline_archived,false,2020-07-09 15:25:37.054,true,3,pHm8C3GMK7LnsWuMcT6sgw==
-mp_pipeline_sales,false,2020-07-09 15:25:37.049,true,4,4sZLZi4xR6kmTh+A8waTCw==
-mp_pipeline_operations,false,2020-07-09 15:25:37.052,true,5,r2LpcB5mggWrgSrGkJERpg==
+pipeline_id,_fivetran_deleted,_fivetran_synced,active,display_order,label,created_at,updated_at
+mp_pipeline_monitoring,FALSE,2020-07-09 15:25:37.051,TRUE,0,83t7Vz1pOW/Fx81JI4o+4g==,2020-07-09 15:25:37.051,2020-07-09 15:25:37.051
+default,FALSE,2020-07-09 15:25:37.053,TRUE,0,FioJAMNdGX8U65mUGCkDwA==,2020-07-09 15:25:37.053,2020-07-09 15:25:37.053
+mp_pipeline_service,FALSE,2020-07-09 15:25:37.024,TRUE,0,eDLdK7glKfd0VRV0G5OcLg==,2020-07-09 15:25:37.024,2020-07-09 15:25:37.024
+mp_pipeline_archived,FALSE,2020-07-09 15:25:37.054,TRUE,3,pHm8C3GMK7LnsWuMcT6sgw==,2020-07-09 15:25:37.054,2020-07-09 15:25:37.054
+mp_pipeline_sales,FALSE,2020-07-09 15:25:37.049,TRUE,4,4sZLZi4xR6kmTh+A8waTCw==,2020-07-09 15:25:37.049,2020-07-09 15:25:37.049
+mp_pipeline_operations,FALSE,2020-07-09 15:25:37.052,TRUE,5,r2LpcB5mggWrgSrGkJERpg==,2020-07-09 15:25:37.052,2020-07-09 15:25:37.052
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply integration_tests/seeds/deal_pipeline_data.csv:7]: That's a good question. I didn't recall doing this, so I did a little investigation. I had opened them in Numbers to add the new columns. I downloaded the lowercase version from the main branch, reopened in Numbers, and lo and behold it had automatically converted all the booleans to uppercase. I looked at some of our other seed data and wonder if the same thing happened to some of them!

File: models/stg_hubspot__ticket.yml (hunk 1 above)
Comment (fivetran-catfritz) [reply models/stg_hubspot__ticket.yml]: Updated!

File: models/stg_hubspot__ticket.yml (hunk 2 above)
Comment (fivetran-catfritz) [reply models/stg_hubspot__ticket.yml]: Yes thank you again. Updated.

File: integration_tests/seeds/deal_pipeline_data.csv (hunk 4 above)
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply integration_tests/seeds/deal_pipeline_data.csv:7]: huh lol thanks for the investigation!

**Review by fivetran-jamie — APPROVED** [review]
lgtm!

**Pull request merged**